include/deal.II/base/tensor_base.h

```diff
@@ -370,6 +394,14 @@ class Tensor<1,dim,Number>
   Tensor<1,dim,Number> &operator = (const Tensor<1,dim,Number> &);
 
   /**
+   * Assignment operator from tensors with different underlying scalar
+   * type. This obviously requires that the @p OtherNumber type is
+   * convertible to @p Number.
+   */
+  template <typename OtherNumber>
+  Tensor<1,dim,Number> &operator = (const Tensor<1,dim,OtherNumber> &);
+
```

Comment:
So I can make the conversion constructor explicit, but assignment operators I can't. So that's what I did.